kernel/kernel-api/src/main/java/io/delta/kernel/internal/stats/FileSizeHistogram.java

```diff
@@ -160,7 +160,15 @@ private static long[] createDefaultBinBoundaries() {
     return boundaries;
   }
 
-  // TODO factory from FileSizeHistogramResult
+  /** Creates a FileSizeHistogram instance from a FileSizeHistogramResult. */
```

Comment:
nit: so `public static FileSizeHistogram fromFileSizeHistogramResult(` already tells me that this method `Creates a FileSizeHistogram instance from a FileSizeHistogramResult.`. IMO we should delete this comment.

However, what would be a helpful comment, is reminding the reader / giving some context on what is a `FileSizeHistogramResult` and how does it relate to this `FileSizeHistogram`? i.e. which code path is this factory involved in?